### ISSUE TITLE
/v2/tasks with Accept: text-plain returns 500 Internal Server Error

### ISSUE THREAD

**Issue by feisuzhu** [mesosphere/marathon#3742]
Apr 12 07:37:00 mesos4 marathon[6207]: [2016-04-12 07:37:00,653] INFO 10.10.81.50 - - [11/Apr/2016:23:37:00 +0000] "GET //mesos4:8080/v2/apps HTTP/1.1" 200 56583 "-" "Java/1.8.0_66" 1 (mesosphere.chaos.http.ChaosRequestLog$$EnhancerByGuice$$ef716d44:qtp1361931872-28)
Apr 12 07:37:00 mesos4 marathon[6207]: [2016-04-12 07:37:00,662] WARN mapping exception to status code (mesosphere.marathon.api.MarathonExceptionMapper$$EnhancerByGuice$$7629671f:qtp1361931872-121)
Apr 12 07:37:00 mesos4 marathon[6207]: java.lang.IndexOutOfBoundsException: Index: 5, Size: 5
Apr 12 07:37:00 mesos4 marathon[6207]: #011at java.util.ArrayList.rangeCheck(ArrayList.java:653) ~[na:1.8.0_66]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at java.util.ArrayList.get(ArrayList.java:429) ~[na:1.8.0_66]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at java.util.Collections$UnmodifiableList.get(Collections.java:1309) ~[na:1.8.0_66]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$$anonfun$appsToEndpointString$2$$anonfun$apply$4$$anonfun$apply$6.apply(EndpointsHelper.scala:38) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$$anonfun$appsToEndpointString$2$$anonfun$apply$4$$anonfun$apply$6.apply(EndpointsHelper.scala:37) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:778) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.Iterator$class.foreach(Iterator.scala:742) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.AbstractIterator.foreach(Iterator.scala:1194) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.MapLike$DefaultValuesIterable.foreach(MapLike.scala:206) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:777) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$$anonfun$appsToEndpointString$2$$anonfun$apply$4.apply(EndpointsHelper.scala:37) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$$anonfun$appsToEndpointString$2$$anonfun$apply$4.apply(EndpointsHelper.scala:34) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:778) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.Iterator$class.foreach(Iterator.scala:742) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.AbstractIterator.foreach(Iterator.scala:1194) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.IterableLike$class.foreach(IterableLike.scala:72) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.AbstractIterable.foreach(Iterable.scala:54) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:777) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$$anonfun$appsToEndpointString$2.apply(EndpointsHelper.scala:34) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$$anonfun$appsToEndpointString$2.apply(EndpointsHelper.scala:20) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:778) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:777) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.EndpointsHelper$.appsToEndpointString(EndpointsHelper.scala:20) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.v2.TasksResource$$anonfun$indexTxt$1.apply(TasksResource.scala:99) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.v2.TasksResource$$anonfun$indexTxt$1.apply(TasksResource.scala:98) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at scala.Option.map(Option.scala:146) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.AuthResource$class.doIfAuthenticated(AuthResource.scala:26) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.v2.TasksResource.doIfAuthenticated(TasksResource.scala:30) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior$$anonfun$invoke$1.apply(DebugConf.scala:56) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.metrics.Metrics.timed(Metrics.scala:27) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior.invoke(DebugConf.scala:55) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.v2.TasksResource.indexTxt(TasksResource.scala:98) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior$$anonfun$invoke$1.apply(DebugConf.scala:56) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.metrics.Metrics.timed(Metrics.scala:27) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior.invoke(DebugConf.scala:55) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at sun.reflect.GeneratedMethodAccessor139.invoke(Unknown Source) ~[na:na]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_66]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_66]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.spi.container.JavaMethodInvokerFactory$1.invoke(JavaMethodInvokerFactory.java:60) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.model.method.dispatch.AbstractResourceMethodDispatchProvider$ResponseOutInvoker._dispatch(AbstractResourceMethodDispatchProvider.java:205) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.model.method.dispatch.ResourceJavaMethodDispatcher.dispatch(ResourceJavaMethodDispatcher.java:75) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.codahale.metrics.jersey.InstrumentedResourceMethodDispatchProvider$TimedRequestDispatcher.dispatch(InstrumentedResourceMethodDispatchProvider.java:30) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.uri.rules.HttpMethodRule.accept(HttpMethodRule.java:302) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1542) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1473) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1419) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1409) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:409) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:540) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:715) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:263) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:178) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:62) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.CacheDisablingFilter.doFilter(CacheDisablingFilter.scala:18) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior$$anonfun$invoke$1.apply(DebugConf.scala:56) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.metrics.Metrics.timed(Metrics.scala:27) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior.invoke(DebugConf.scala:55) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterDefinition.doFilter(FilterDefinition.java:163) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:58) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.CORSFilter.doFilter(CORSFilter.scala:46) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior$$anonfun$invoke$1.apply(DebugConf.scala:56) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.metrics.Metrics.timed(Metrics.scala:27) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.DebugModule$MetricsBehavior.invoke(DebugConf.scala:55) ~[marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterDefinition.doFilter(FilterDefinition.java:163) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:58) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.LimitConcurrentRequestsFilter.pass(LimitConcurrentRequestsFilter.scala:32) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.LimitConcurrentRequestsFilter$$anonfun$3$$anonfun$apply$2.apply(LimitConcurrentRequestsFilter.scala:15) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.LimitConcurrentRequestsFilter$$anonfun$3$$anonfun$apply$2.apply(LimitConcurrentRequestsFilter.scala:15) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.LimitConcurrentRequestsFilter.doFilter(LimitConcurrentRequestsFilter.scala:36) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterDefinition.doFilter(FilterDefinition.java:163) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:58) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at mesosphere.marathon.api.LeaderProxyFilter.doFilter(LeaderProxyFilter.scala:126) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterDefinition.doFilter(FilterDefinition.java:163) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:58) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:118) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:113) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1158) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1090) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:240) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:109) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.Server.handle(Server.java:517) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:308) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:242) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:261) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:75) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceAndRun(ExecuteProduceConsume.java:213) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:147) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:654) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:572) [marathon:0.15.3]
Apr 12 07:37:00 mesos4 marathon[6207]: #011at java.lang.Thread.run(Thread.java:745) [na:1.8.0_66]

**Comment by feisuzhu** [comment]
Marathon v0.15.3

We have changed app conf to add another service port.

During deployment when old instances(5 service ports) coexist with new instances(6 service ports), /v2/tasks will fail.

**Comment by pierlo-upitup** [comment]
Hi @feisuzhu could you paste the app definition JSON for the app that's causing this? Also, which version of Mesos are you running this on? Thanks!

**Comment by tylermarshall** [comment]
Hey @pierlo-upitup , I have also run into this problem and am able to reproduce it. The relevant piece from my app definition is the container section. I am able to deploy my app successful with the below settings:
```
  "container": {
    "type": "DOCKER",
    "volumes": [ ],
    "docker": {
      "image": "nginx:1.9",
      "network": "BRIDGE",
      "portMappings": [
        {
          "containerPort": 80,
          "hostPort": 0,
          "servicePort": 22761,
          "protocol": "tcp"
        }
      ]
    }
  }
```
Now the interesting part is when I change the port mappings array to the following:
```
  "container": {
    "type": "DOCKER",
    "volumes": [ ],
    "docker": {
      "image": "nginx:1.9",
      "network": "BRIDGE",
      "portMappings": [
        {
          "containerPort": 81,
          "hostPort": 0,
          "servicePort": 22761,
          "protocol": "tcp"
        },
        {
          "containerPort": 9200,
          "hostPort": 0,
          "servicePort": 22762,
          "protocol": "tcp"
        }
      ]
    }
  }
```
It doesn't matter if my application comes up healthy or not. The steps above will trigger this error.
```
curl -sSfLk  -m 10 -H 'Accept: text/plain' http://<marathon url>/v2/tasks
curl: (22) The requested URL returned error: 500 Server Error
```

**Comment by gkleiman** [comment]
Fixed by 01d822ec64319922e3d9d68305317b5faccbc289
